airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/integrations/source/relationaldb/state/GlobalStateManager.kt

```diff
@@ -64,7 +64,17 @@ class GlobalStateManager(
         // Populate global state
         val globalState = AirbyteGlobalState()
         globalState.sharedState = Jsons.jsonNode(cdcStateManager.cdcState)
-        globalState.streamStates = StateGeneratorUtils.generateStreamStateList(pairToCursorInfoMap)
+        // If stream state exists in the global manager, it should be used to reflect the partial
+        // states of initial loads.
+        if (
+            cdcStateManager.rawStateMessage?.global?.streamStates != null &&
+                cdcStateManager.rawStateMessage.global?.streamStates?.size != 0
+        ) {
+            globalState.streamStates = cdcStateManager.rawStateMessage.global.streamStates
```

Comment:
what does this line do?

Reply:
This line is needed to make sure that the state that the CDC/debezium path emits the partial states from the initial load. E.g. if Stream A has completed until `pk_val`, we want to make sure this is reflected in the state emitted by the CDC iterators. 

Previously, the CDC state emission logic emitted any streams configured as complete which would result in data loss in the WASS case (as Stream A would not complete again)